File: quitionanalyzer.cpp
#include <iostream>
#include <string>
#include <cmath>
#include <ctime>
#include <cstdlib>
using namespace std;

#define Max_try 4
#define numb_of_word 7
#define numb_of_ques 5
#define numb_face 4

struct Animal {
    double word[numb_of_word];
    string name;
};

struct User {
    string username;
    string password;
    double word[numb_of_word]{0};
    int attempts = 0;
    bool locked = false;
};

Animal animals[numb_face] = {
    {{1.0/15,1.0/15,5.0/15,0,3.0/15,4.0/15,1.0/15},"DOG"},
    {{1.0/10,1.0/10,4.0/10,1.0/10,2.0/10,3.0/10,2.0/10},"CAT"},
    {{0.1,0.2,0.3,0.1,0.2,0.4,0.1},"RABBIT"},
    {{0.3,0.3,0.1,0.2,0.2,0.2,0.1},"LION"}
};

double calculate(double u[], double a[]) {
    double sum = 0;
    for(int i=0;i<numb_of_word;i++) sum += pow(u[i]-a[i],2);
    return sum;
}

bool admin_unlock(User &u){
    string id, pass;
    cout<<"ADMIN ID: ";
    cin>>id;
    cout<<"ADMIN PASS: ";
    cin>>pass;
    if(id=="admin" && pass=="6666"){
        u.locked=false;
        u.attempts=0;
        cout<<"Account Unlocked\n";
        return true;
    }
    cout<<"Invalid Admin Login\n";
    return false;
}

bool login(User &u){
    while(u.attempts<Max_try){
        cout<<"Enter Username: ";
        cin>>u.username;
        cout<<"Enter Password: ";
        cin>>u.password;

        if(u.username.size()==8 && u.password==u.username.substr(4,4)){
            cout<<"Login Successful\n";
            return true;
        }

        cout<<"Login Failed\n";
        u.attempts++;

        if(u.attempts==Max_try){
            u.locked=true;
            cout<<"Account Locked. Admin Required.\n";
            while(!admin_unlock(u));
            return login(u);
        }
    }
    return false;
}

void shuffleQ(string q[], string opt[][4], int score[numb_of_ques][4][numb_of_word]){
    srand(time(0));
    for(int i=0;i<numb_of_ques;i++){
        int j=rand()%numb_of_ques;
        swap(q[i],q[j]);
        for(int k=0;k<4;k++){
            swap(opt[i][k],opt[j][k]);
            for(int x=0;x<numb_of_word;x++)
                swap(score[i][k][x],score[j][k][x]);
        }
    }
}

void take(User &u){
    string q[numb_of_ques]={
        "1. You find a puzzle, you choose to:",
        "2. When facing a challenge, you prefer to:",
        "3. In a social group, you are the one who:",
        "4. On a rainy day, you would:",
        "5. You are known for being:"
    };

    string opt[numb_of_ques][4]={
        {"A. Solve it alone","B. Ask help","C. Observe","D. Ignore"},
        {"A. Lead","B. Plan","C. Risk","D. Wait"},
        {"A. Lead","B. Organize","C. Support","D. Alone"},
        {"A. Read","B. Walk","C. Socialize","D. Build"},
        {"A. Outgoing","B. Reflective","C. Independent","D. Kind"}
    };

    int score[numb_of_ques][4][numb_of_word]={
        {{2,0,0,0,0,0,1},{1,2,0,0,0,0,2},{0,0,2,0,0,0,1},{0,0,0,2,0,0,0}},
        {{2,0,0,0,0,0,1},{1,1,0,0,0,1,2},{0,0,2,0,2,0,0},{0,0,0,2,0,0,0}},
        {{2,1,0,0,0,1,1},{1,2,0,0,1,0,2},{0,0,2,0,0,0,0},{0,0,0,1,1,2,1}},
        {{0,0,1,0,0,2,2},{0,1,0,2,0,1,2},{1,0,0,1,1,0,0},{2,0,0,0,1,0,1}},
        {{1,0,0,0,2,0,1},{1,2,0,0,0,0,0},{0,1,2,0,0,0,1},{0,0,1,2,0,1,2}}
    };

    shuffleQ(q,opt,score);

    for(int i=0;i<numb_of_ques;i++){
        cout<<q[i]<<endl;
        for(int j=0;j<4;j++) cout<<opt[i][j]<<endl;
        char ans;
        cin>>ans;
        int idx=ans-'A';
        for(int k=0;k<numb_of_word;k++) u.word[k]+=score[i][idx][k];
    }
}

void face(User &u){
    double best=1e18;
    int idx=-1;
    for(int i=0;i<numb_face;i++){
        double v=calculate(u.word,animals[i].word);
        if(v<best){best=v; idx=i;}
    }
    cout<<"Your Animal Face: "<<animals[idx].name<<endl;
}

int main(){
    User u;
    if(login(u)){
        do{
            take(u);
            face(u);
            cout<<"Again? (Y/N): ";
            char c; cin>>c;
            if(c=='N'||c=='n') break;
        }while(true);
    }
}

File: quitionanalyzer.py
import random
import math

Max_try = 4
numb_of_word = 7
numb_of_ques = 5
numb_face = 4

class Animal:
    def __init__(self, word, name):
        self.word = word
        self.name = name

class User:
    def __init__(self, username="", password=""):
        self.username = username
        self.password = password
        self.word = [0] * numb_of_word
        self.attempts = 0

animals = [
    Animal([1/15, 1/15, 5/15, 0/15, 3/15, 4/15, 1/15], "DOG"),
    Animal([1/10, 1/10, 4/10, 1/10, 2/10, 3/10, 2/10], "CAT"),
    Animal([0.1, 0.2, 0.3, 0.1, 0.2, 0.4, 0.1], "RABBIT"),
    Animal([0.3, 0.3, 0.1, 0.2, 0.2, 0.2, 0.1], "LION")
]

predefined_users = [
    User("admin1234", "1234"),
    User("demo1234", "1234")
]

def calculate(user, animal):
    total = 0
    for i in range(numb_of_word):
        total += (user[i] - animal[i]) ** 2
    return total

def login(user):
    attempts = 0
    while attempts < Max_try:
        u = input("Enter Username: ")
        p = input("Enter Password: ")

        for udata in predefined_users:
            if u == udata.username and p == udata.username[-4:]:
                user.username = udata.username
                user.password = udata.password
                print("Login Successful! Welcome", user.username)
                return 1

        print("Login Failed")
        attempts += 1
        if attempts == Max_try:
            print("Account Locked. Contact Admin.")
            return 0
    return 0

def shuffle_questions(questions, options, score):
    for i in range(numb_of_ques):
        j = random.randint(0, numb_of_ques - 1)
        if i != j:
            questions[i], questions[j] = questions[j], questions[i]
            options[i], options[j] = options[j], options[i]
            score[i], score[j] = score[j], score[i]

def take_quiz(user):
    questions = [
        "1. You find a puzzle, you choose to: ",
        "2. When facing a challenge, you prefer to: ",
        "3. In a social group, you're often the one who: ",
        "4. On a rainy day, you would likely: ",
        "5. You are known for being: "
    ]

    options = [
        ["A. Solve it alone","B. Seek help","C. Observe first","D. Let others solve"],
        ["A. Lead the way","B. Plan carefully","C. Take risks","D. Wait for solution"],
        ["A. Take charge","B. Organize","C. Support","D. Enjoy space"],
        ["A. Read","B. Walk","C. Socialize","D. Build"],
        ["A. Outgoing","B. Reflective","C. Independent","D. Compassionate"]
    ]

    score = [
        [[2,0,0,0,0,0,1],[1,2,0,0,0,0,2],[0,0,2,0,0,0,1],[0,0,0,2,0,0,0]],
        [[2,0,0,0,0,0,1],[1,1,0,0,0,1,2],[0,0,2,0,2,0,0],[0,0,0,2,0,0,0]],
        [[2,1,0,0,0,1,1],[1,2,0,0,1,0,2],[0,0,2,0,0,0,0],[0,0,0,1,1,2,1]],
        [[0,0,1,0,0,2,2],[0,1,0,2,0,1,2],[1,0,0,1,1,0,0],[2,0,0,0,1,0,1]],
        [[1,0,0,0,2,0,1],[1,2,0,0,0,0,0],[0,1,2,0,0,0,1],[0,0,1,2,0,1,2]]
    ]

    shuffle_questions(questions, options, score)

    for i in range(numb_of_ques):
        print(questions[i])
        for op in options[i]:
            print(op)
        ans = input("Enter Choice (A-D): ").upper()
        idx = ord(ans) - ord('A')
        for j in range(numb_of_word):
            user.word[j] += score[i][idx][j]

def find_face(user):
    minv = float('inf')
    name = ""
    for ani in animals:
        v = calculate(user.word, ani.word)
        if v < minv:
            minv = v
            name = ani.name
    print("Your Animal Face:", name)

def try_again():
    c = input("Try Again? (Y/N): ")
    if c in ['Y','y']:
        return 1
    return 0

def main():
    user = User()
    if login(user):
        while True:
            user.word = [0] * numb_of_word
            take_quiz(user)
            find_face(user)
            if not try_again():
                break

main()
